nautobot/core/tables.py

```diff
@@ -140,13 +141,55 @@ def __init__(self, *args, user=None, **kwargs):
                         select_fields.append("__".join(select_path))
                     elif prefetch_path:
                         prefetch_fields.append("__".join(prefetch_path))
-            logger.debug(
-                "Applying .select_related(%s).prefetch_related(%s) to %s QuerySet",
-                select_fields,
-                prefetch_fields,
-                self.data.data.model.__name__,
-            )
-            self.data.data = self.data.data.select_related(*select_fields).prefetch_related(*prefetch_fields)
+
+            if select_fields:
+                # Django doesn't allow .select_related() on a QuerySet that had .values()/.values_list() applied, or
+                # one that has had union()/intersection()/difference() applied.
+                # We can detect and avoid these cases the same way that Django itself does.
+                if queryset._fields is not None:
+                    logger.debug(
+                        "NOT applying select_related(%s) to %s QuerySet as it includes .values()/.values_list()",
+                        select_fields,
+                        model.__name__,
+                    )
+                elif queryset.query.combinator:
+                    logger.debug(
+                        "NOT applying select_related(%s) to %s QuerySet as it is a combinator query",
+                        select_fields,
+                        model.__name__,
+                    )
+                else:
+                    logger.debug("Applying .select_related(%s) to %s QuerySet", select_fields, model.__name__)
+                    # Belt and suspenders - we should have avoided any error cases above, but be safe anyway:
+                    try:
+                        queryset = queryset.select_related(*select_fields)
+                    except (TypeError, ValueError, NotSupportedError) as exc:
+                        logger.warning(
+                            "Unexpected error when trying to .select_related() on %s QuerySet: %s",
+                            model.__name__,
+                            exc,
+                        )
+
+            if prefetch_fields:
+                if queryset.query.combinator:
```

Comment:
`values()/values_list()` are allowed for `prefetch_related()` right?

Reply:
Correct.